### PR TITLE
Feature SDK analytics

### PR DESCRIPTION
**Pull request by adimiz1**
### Brief Summary of Changes
Add feature SDK analytics support

#### What does this PR address?
- [ ] GitHub issue (Add reference - #XX)
- [ ] Refactoring
- [x] New feature
- [ ] Bug fix
- [x] Adds more tests

#### Are tests included?
- [x] Yes
- [ ] No

#### Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I ran the full test suite before pushing the changes and all the tests pass.

### CODE REVIEW

File: core/src/main/java/com/cloudinary/android/MediaManager.java [hunk 1]
```diff
@@ -232,13 +244,6 @@ public Cloudinary getCloudinary() {
      */
     public Url url() {
         Url url = cloudinary.url();
-
-        // set https as default for android P and up - in P the default policy fails all http
-        // requests
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
```
Comment (patrick-tolosa) [inline core/src/main/java/com/cloudinary/android/MediaManager.java:238]: why was this removed in this PR?

**Review by patrick-tolosa — APPROVED** [review]
Approved, but I'm not clear why we removed a piece of code there for HTTPS.

File: core/src/main/java/com/cloudinary/android/MediaManager.java (hunk 1 above)
Comment (adimiz1) [reply core/src/main/java/com/cloudinary/android/MediaManager.java:238]: It was not removed, it was moved to the MediaManager `init` function instead of the `URL` init

File: core/src/main/java/com/cloudinary/android/MediaManager.java [hunk 2]
```diff
@@ -86,6 +90,17 @@ private MediaManager(@NonNull Context context, @Nullable SignatureProvider signa
...
         }
 
+        if (cloudinary.config.analytics == null) {
+            cloudinary.config.analytics = true;
+        }
+        cloudinary.setAnalytics(new Analytics("F", VERSION, techVersion));
+
+        // set https as default for android P and up - in P the default policy fails all http
+        // requests
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
```
Comment (patrick-tolosa) [inline core/src/main/java/com/cloudinary/android/MediaManager.java:100]: This was removed right?

File: core/src/main/java/com/cloudinary/android/MediaManager.java (hunk 1 above)
Comment (patrick-tolosa) [reply core/src/main/java/com/cloudinary/android/MediaManager.java:238]: See my comment above, now it needs to be removed:)
